test/fashion-mnist/infer.py

```diff
@@ -0,0 +1,115 @@
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torchvision
+import torchvision.transforms as transforms
+
+# Define the neural network model
+class Net(nn.Module):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.fc1 = nn.Linear(784, 256)
+        self.fc2 = nn.Linear(256, 128)
+        self.fc3 = nn.Linear(128, 10)
+
+    def forward(self, x):
+        x = x.view(x.size(0), -1)
+        x = torch.relu(self.fc1(x))
+        x = torch.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+## IMPROVE
+from improve import framework as frm
+import candle
+from pathlib import Path
+
+filepath = Path(__file__).resolve().parent
+
+
+# Part 3: Model Testing
+## IMPROVE
+def run(params):
+##
+
+   # Need to get testloader from Part 1.
+   transform = transforms.Compose([
+      transforms.ToTensor(),
+      transforms.Normalize((0.5,), (0.5,))
+   ])
+   
+   # Get the data directory, batch size and other hyperparameters from params 
+   ##IMPROVE 
+   batch_size = params["batch_size"]
+   learning_rate = params["learning_rate"]
+   momentum = params["momentum"]
+   dataset_dir = params["data_dir"]
+   
+   # NOTE: using false now for data loading
+   testset = torchvision.datasets.FashionMNIST(root=dataset_dir, train=False, download=False, transform=transform)
+   testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size, shuffle=True)
+
+   # Check if GPU is available, else use CPU
+   device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+   
+   # Create a neural network model
+   model = Net().to(device)
+
+   # Define optimizer
+   optimizer = optim.SGD(model.parameters(), lr=learning_rate, momentum=momentum)
```

Comment:
Same here

Reply:
same as above :) we can get the model weights in some other fashion, do you know how else to get the model weights to perform inferece? -  where optimizer or learning rate is not needed, it is a minor thing and can be ignored, IMO. The overall logic is to get the model weights and infer on what was done in the training step.

Reply:
I agree, get the model weights and infer on any input data in the input data directory. The model weights should be located there as well. The outputs of training are model weights and learning metrics.